zjit/src/bitset.rs

```diff
@@ -0,0 +1,70 @@
+type Entry = u128;
+
+// TODO(max): Make a `SmallBitSet` and `LargeBitSet` and switch between them if `num_bits` fits in
+// `Entry`.
+pub struct BitSet<T: Into<usize> + Copy> {
+    storage: Vec<Entry>,
+    num_bits: usize,
+    phantom: std::marker::PhantomData<T>,
+}
+
+impl<T: Into<usize> + Copy> BitSet<T> {
+    pub fn with_capacity(num_bits: usize) -> Self {
+        // +1 because we are rounding down
+        let num_entries = num_bits / (Entry::BITS as usize) + 1;
```

Comment:
Does it make sense to store `Entry::BITS as usize` in a constant?

Reply:
Also, maybe this is simpler?
```suggestion
        let num_entries = num_bits.div_ceil(Entry::BITS as usize);
```